drivers/accel/adxl355/iio_adxl355.c

```diff
@@ -0,0 +1,1032 @@
+		if (ret)
+			return ret;
+
+		// Update 3db frequency table for new ODR value
+		return adxl355_iio_fill_3db_frequency_table(iio_adxl355);
+
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Handles the read request for filter_high_pass_3db_frequency_available
+ * 		  attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl355_iio_read_hpf_available(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2];
+	char buffer[20];
+	struct adxl355_iio_dev *iio_adxl355;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl355 = (struct adxl355_iio_dev *)dev;
+
+	switch (channel->type) {
+	case IIO_ACCEL:
+		// Remove data existing in the buffer
+		strcpy(buf, "");
+
+		// Go through the values in HPF table for the current ODR and add them to the buffer
+		for (uint8_t i = 0; i < NO_OS_ARRAY_SIZE(iio_adxl355->adxl355_hpf_3db_table);
+		     i++) {
+
+			vals[0] = iio_adxl355->adxl355_hpf_3db_table[i][0];
+			vals[1] = iio_adxl355->adxl355_hpf_3db_table[i][1];
+			iio_format_value(buffer, len, IIO_VAL_INT_PLUS_MICRO, 2, vals);
+			strcat(buf, buffer);
+			strcat(buf, " ");
+		}
+		return strlen(buf);
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Handles the read request for sampling_frequency_available attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl355_iio_read_samp_freq_avail(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2];
+	char buffer[20];
+
+	switch (channel->type) {
+	case IIO_ACCEL:
+		// Remove data existing in the buffer
+		strcpy(buf, "");
+
+		// Go through the values in ODR table and add them to the buffer
+		for (uint8_t i = 0; i < NO_OS_ARRAY_SIZE(adxl355_iio_odr_table); i++) {
+			vals[0] = adxl355_iio_odr_table[i][0];
+			vals[1] = adxl355_iio_odr_table[i][1];
+			iio_format_value(buffer, sizeof(buffer), IIO_VAL_INT_PLUS_MICRO, 2, vals);
+			strcat(buf, buffer);
```

Comment:
why do we need to add this whitespace after iio_format_value ? should we add it inside iio_format_value perhaps ?

Reply:
The whitespace is added because, in this case,  there is more than one element element returned in the buffer. 
The buffer will contain the list of available sampling frequencies, which are added one by one in the returned buffer and are delimitated by the white space. I don't think it is needed to be added to iio_format_value just for this list usage scenario.

Reply:
ok, makes sense